eng/build.yml

```diff
@@ -1,14 +1,4 @@
 steps:
-- task: UseDotNet@2
-  inputs:
-    version: 3.1.404
-  displayName: Install .NET Core SDK 3.1.404
-
-- task: UseDotNet@2
-  inputs:
-    version: 5.0.401
-  displayName: Install .NET Core SDK 5.0.401
-
```

Comment:
@Bertk Thanks a lot for your time-consuming contributions 🙏 . I've seen your comment in [https://github.com/coverlet-coverage/coverlet/issues/1511#issuecomment-1663517315](https://github.com/coverlet-coverage/coverlet/issues/1511#issuecomment-1663517315) regarding this topic but PR was already merged at that time. I think you are right that we don't need them anymore. Especially if we migrate all the test projects to `net6.0` like you have done in this PR.
Personally I'm not sure if we shouldn't keep 1-2 legacy `netcoreapp3.1` projects to have a wider range of test scenarios regarding instrumentation... What do you think @MarcoRossignoli?
Thanks again, I really like the changes in this PR and will try to do a proper review in the next days.

Reply:
I suggest to test and ship only supported versions announced here https://learn.microsoft.com/en-us/lifecycle/products/microsoft-net-and-net-core

Reply:
This is shows ***supported versions*** much better :wink:
https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core